ipi/engine/motion/driven_dynamics.py

```diff
@@ -0,0 +1,177 @@
+        # If they are not the same, then there is a bug in the code
+        dt = abs(self.ensemble.time - self.integrator.mts_time)
+        if dt > 1e-12:
+            raise ValueError(
+                "The time at which the Electric Field is evaluated is not properly updated!"
+            )
+
+
+dproperties(DrivenDynamics, ["dt", "nmts", "splitting", "ntemp"])
+
+
+class EDAIntegrator(DummyIntegrator):
+    """Integrator object for simulations using the Electric Dipole Approximation (EDA)
+    when an external electric field is applied.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def bind(self, motion):
+        """bind variables"""
+        super().bind(motion)
+
+        self._time = self.eda._time
+        self._mts_time = self.eda._mts_time
+
+        dep = [
+            self._time,
+            self._mts_time,
+            self.eda.Born_Charges._bec,
+            self.eda.Electric_Field._Efield,
+        ]
+        self._EDAforces = depend_array(
+            name="EDAforces",
+            func=self._eda_forces,
+            value=np.zeros((self.beads.nbeads, self.beads.natoms * 3)),
+            dependencies=dep,
+        )
+        pass
+
+    def pstep(self, level=0):
+        """Velocity Verlet momentum propagator."""
+        self.beads.p += self.EDAforces * self.pdt[level]
+        if dstrip(self.mts_time) == dstrip(self.time):
+            # it's the first time that 'pstep' is called
+            # then we need to update 'mts_time'
+            self.mts_time += dstrip(self.dt)
+            # the next time this condition will be 'False'
+            # so we will avoid to re-compute the EDAforces
+        pass
+
+    def _eda_forces(self):
+        """Compute the EDA contribution to the forces, i.e. `q_e Z^* @ E(t)`"""
+        Z = dstrip(self.eda.Born_Charges.bec)  # tensor of shape (nbeads,3xNatoms,3)
+        E = dstrip(self.eda.Electric_Field.Efield)  # vector of shape (3)
+        forces = Constants.e * Z @ E  # array of shape (nbeads,3xNatoms)
+        return forces
+
+    def step(self, step=None):
+        if len(self.nmts) > 1:
+            raise ValueError(
+                "EDAIntegrator is not implemented with the Multiple Time Step algorithm (yet)."
+            )
+        # This should call 'NVEIntegrator.step' since 'self' should be an instance of 'EDANVEIntegrator' and not of 'EDAIntegrator'
+        super().step(step)
+
+
+dproperties(EDAIntegrator, ["EDAforces", "mts_time", "time"])
+
+
+class EDANVEIntegrator(EDAIntegrator, NVEIntegrator):
+    """Integrator object for simulations with constant Number of particles, Volume, and Energy (NVE)
+    using the Electric Dipole Approximation (EDA) when an external electric field is applied.
+    """
+
+    def pstep(self, level):
+        # NVEIntegrator does not use 'super()' within 'pstep'
+        # then we can not use 'super()' here.
+        # We need to call the 'pstep' methods explicitly.
+        NVEIntegrator.pstep(
```

Comment:
are you misusing super? should be possible to call the parent function, see the NPTIntegrator
```
        super(NPTIntegrator, self).pstep(level)
```

Reply:
Still don't understand the comment. You mean that if you write here 
```
super(EDANVEIntegrator, self).pstep(level) 
``` 
it doesn't work?